desktop/core/src/desktop/js/apps/storageBrowser/StorageFilePage/StorageFilePage.tsx

```diff
@@ -20,98 +20,175 @@ import './StorageFilePage.scss';
 import { i18nReact } from '../../../utils/i18nReact';
 import Button, { PrimaryButton } from 'cuix/dist/components/Button';
 import { getFileMetaData } from './StorageFilePage.util';
-import { DOWNLOAD_API_URL } from '../../../reactComponents/FileChooser/api';
+import { DOWNLOAD_API_URL, SAVE_FILE_API_URL } from '../../../reactComponents/FileChooser/api';
 import huePubSub from '../../../utils/huePubSub';
+import useSaveData from '../../../utils/hooks/useSaveData';
+import {
+  EDITABLE_FILE_FORMATS,
+  SUPPORTED_FILE_EXTENSIONS,
+  SupportedFileTypes
+} from '../../../utils/constants/storageBrowser';
+import { Spin } from 'antd';
 
 const StorageFilePage = ({ fileData }: { fileData: PathAndFileData }): JSX.Element => {
   const { t } = i18nReact.useTranslation();
   const [isEditing, setIsEditing] = React.useState(false);
   const [fileContent, setFileContent] = React.useState(fileData.view?.contents);
   const fileMetaData = useMemo(() => getFileMetaData(t, fileData), [t, fileData]);
 
+  const { loading: isSaving, save } = useSaveData(SAVE_FILE_API_URL, {
+    onSuccess: () => {
+      setIsEditing(false);
+    }
+  });
+
   const handleEdit = () => {
     setIsEditing(true);
   };
 
+  const handleCancel = () => {
+    setIsEditing(false);
+    setFileContent(fileData.view?.contents);
+  };
+
+  const handleSave = () =>
+    save({
```

Comment:
Now if we are establishing a pattern it is reasonable be a bit more picky than usual :-) 

I thought about this during the other PR review. Doesn't it make more sense to handle the save success and failure here instead of at the top where we get the save function from the hook? That way the logic is grouped together.

`save({...}).then(()=>{do stuff}).catch(...)` 

or if the error handling is done by the hook like in this example
```
const success = await save({...});
if (success ) {...}
```

Reply:
Based on our conversation earlier, I researched a bit and found that the best flexible/scalable design pattern for this hook would be to have `onSuccess` and `onError` hook be placed on both the places, one in the hook initialisation and other while making the save function call.
I have changed the implemented for the hook as well.

Let me know your thoughts on this.